src/components/Deferred.tsx

```diff
@@ -0,0 +1,38 @@
+import type {PropsWithChildren} from 'react';
+import {startTransition, use, useLayoutEffect, useState} from 'react';
+
+/**
+ * This is a wrapper component that allows us to defer rendering children and do it in the background with the use of startTransition.
+ * Caution: To achieve performance benefits from this component you have to wrap it in a Suspense component.
+ */
+function Deferred({children}: PropsWithChildren) {
+    const [{promise, resolve}] = useState(() => {
+        let resolver!: () => void;
```

Comment:
NAB, but I figured out while reviewing another PR that [it's really easy to polyfill `Promise.withResolvers`](https://github.com/Expensify/App/pull/68190#discussion_r2302571182) in this repo, and that makes this a bit cleaner:

```ts
const [{promise, resolve}] = useState(Promise.withResolvers());
```